### PR TITLE
Encode template data inside send functions

### PR DESCRIPTION
**Pull request by thermech**
As suggested by @lessless on https://github.com/ex-aws/ex_aws_ses/pull/5#pullrequestreview-209787295, encode the value inside the method

### CODE REVIEW

File: lib/ex_aws/ses.ex [hunk 1]
```diff
@@ -169,9 +169,9 @@ defmodule ExAws.SES do
       |> Map.merge(format_member_attribute(:reply_to_addresses, opts[:reply_to]))
       |> Map.merge(format_tags(opts[:tags]))
       |> Map.merge(format_dst(dst))
-      |> Map.put_new("Source", src)
-      |> Map.put_new("Template", template)
-      |> Map.put_new("TemplateData", template_data)
+      |> Map.put("Source", src)
+      |> Map.put("Template", template)
+      |> Map.put("TemplateData", format_template_data(template_data))
```
Comment (Boubalou) [inline lib/ex_aws/ses.ex:174]: Why did you change all `put_new` for `put`?

Comment (thermech) [reply lib/ex_aws/ses.ex:174]: Because the value has to be replace if already exist. For exemple: `TemplateData` was not replaced by `format_template_data(template_data)` because it was already in the map

**Comment by thermech** [comment]
@remiprev @simonprev plz

File: lib/ex_aws/ses.ex [hunk 2]
```diff
@@ -279,9 +279,9 @@ defmodule ExAws.SES do
     |> flatten_attrs(root)
   end
 
-  defp format_default_template_data(nil), do: "{}"
+  defp format_template_data(nil), do: "{}"
 
-  defp format_default_template_data(default_template_data), do: default_template_data
+  defp format_template_data(template_data), do: template_data |> Poison.encode!()
```
Comment (simonprev) [inline lib/ex_aws/ses.ex]: For single value piping we usually do:

```suggestion
  defp format_template_data(template_data), do: Poison.encode!(template_data)
```

There is a credo rule for this, maybe we could take the time to add it to our fork?

**Review by simonprev — APPROVED** [review]

File: lib/ex_aws/ses.ex (hunk 2 above)
Comment (thermech) [reply lib/ex_aws/ses.ex]: I wont enforce the rule as I want to fork it back to origin, but I'll apply the fixes on my PR
